transform_aes.c

```diff
@@ -182,25 +221,46 @@ static int transop_decode_aes( n2n_trans_op_t * arg,
 
                 AES_cbc_encrypt( (inbuf + TRANSOP_AES_PREAMBLE_SIZE),
                                  assembly, /* destination */
-                                 len, 
+                                 len,
                                  &(priv->dec_key),
                                  dec_ivec, AES_DECRYPT);
 
                 /* last byte is how much was padding: max value should be
                  * AES_BLOCKSIZE-1 */
-                padding = assembly[ len-1 ] & 0xff; 
+                padding = assembly[ len-1 ] & 0xff;
 
-                if ( len >= padding)
+                if ( len >= (padding + TRANSOP_AES_COMPRESSION_SIZE) )
                 {
                     /* strictly speaking for this to be an ethernet packet
                      * it is going to need to be even bigger; but this is
                      * enough to prevent segfaults. */
                     traceEvent(TRACE_DEBUG, "padding = %u", padding);
                     len -= padding;
-
-                    memcpy( outbuf, 
-                            assembly, 
-                            len);
+		    len -= TRANSOP_AES_COMPRESSION_SIZE;
+
+		    /* decompress if necessary */
```

Comment:
`#ifdef COMPRESSION_ENABLED`?

Reply:
For the sake of being able to decompress packets even if a node does not actively makes use of compression, I did not def'ed it out by intent. This allows mixed environments where nodes with some weaker CPU shall not compress - but should be able to decompress in case they are talked to.

But we can switch to the "either ... or"-way which would force all nodes of one network to share the same compression setting. Please advice.

Reply:
This too seems to be a design decision.

Reply:
Done.